### PR TITLE
Update language level df

### LINKED ISSUE #19: language_level_df and drop_duplicate_tips_random
`drop_duplicate_tips_random()` in #15  drops all but one tip in trees with the same glottocode. It has an argument `merge_dialects`which can be set to TRUE, and if so it replaces tips with the glottocode of their parent language and then does the duplicate dropping. If FALSE, it just takes the glottocodes of the dialects straight and keeps all but one if there are duplicates of that dialect.

`language_level_df()` currently doesn't have this option, it just merges dialects whether you want or not. I think that's probably fine, but I thought I'd check if anyone has a use case where they want to drop duplicates but _NOT_ merge dialects into one language.

### PR DESCRIPTION
**Pull request by HedvigS**
* some clean up (remove drop_question_mark argument)
* set it up so that ValueTable is first match to LanguageTable via Language_ID and then Glottocodes is used for reduction, instead of assuming that Language_IDs are Glottocodes.
* add LanguageTable2 as an optional argument for those cldf-datasets that don't have Language_ID or Language_level_ID in their LanguageTable and therefore needs an additional table
* change the test, because I found it hard to parse what was happening so I change to a simple count of how many unique glottocodes remain at the end.

closes #19 and #23

### CODE REVIEW

**Comment by HedvigS** [comment]
In addition, changed name from `language_level_df` to `reduce_ValueTable_to_unique_glottocodes` to make it more transperent.

**Comment by HedvigS** [comment]
@xrotwang just reminded me of datasets like APiCs where it's possible to have more than one Value per Language_ID and Parameter_ID in the ValueTable. I've not adjusted the function for this yet, but in the meantime I added n a Note and an error for this.

**Comment by HedvigS** [comment]
> @xrotwang just reminded me of datasets like APiCs where it's possible to have more than one Value per Language_ID and Parameter_ID in the ValueTable. I've not adjusted the function for this yet, but in the meantime I added n a Note and an error for this.

Sometimes, all you gotta do is have a little break and a think. I thought out a solution while biking home from work, committing in a sec :)!

**Comment by HedvigS** [comment]
Okay, I've modified the function now so that if it encounters a ValueTable similar to the one in APiCS, it does something good. It can't handle all those kinds of situations, but it can handle APICS and others with the same columns in ValueTable.

**Comment by xrotwang** [comment]
I don't think that piling up special case handling in this function is a good idea. Now it handles the APiCS case, expecting non-CLDF-standard columns "Frequency" and "Confidence" with particular semantics. But APiCS is about pidgins and creoles - i.e. languoids where Glottolog classification is somewhat different to the other families and "collapsing to language level" less well specified as for other cases.

So, as I said elsewhere - I think "rgrambank" should be targeted to handling access to and analysis of Grambank and not striving for ill-defined generality.

**Comment by HedvigS** [comment]
> I don't think that piling up special case handling in this function is a good idea. Now it handles the APiCS case, expecting non-CLDF-standard columns "Frequency" and "Confidence" with particular semantics. But APiCS is about pidgins and creoles - i.e. languoids where Glottolog classification is somewhat different to the other families and "collapsing to language level" less well specified as for other cases.
> 
> So, as I said elsewhere - I think "rgrambank" should be targeted to handling access to and analysis of Grambank and not striving for ill-defined generality.

Okay, I hear you.

To me, this function is now very useful and I'd love it if it's in rgrambank. I realise the APiCS handling is a bit hacky, but I'm okay with it at this stage and it doesn't interfere with other things I want to do.

**Comment by HedvigS** [comment]
I don't know anyone else who is developing code for this purpose, so I'll make a stab at it myself for cases I have encountered or can envisage in short term and then we'll see what @SimonGreenhill says.

**Comment by HedvigS** [comment]
I've merged in main and resolved conflicts and updated namespace.
